pkg/lib/validate/collection.go

```diff
@@ -0,0 +1,9 @@
+package validate
+
+func IsEmpty[T any](s []T) bool {
+	return len(s) == 0
+}
+
+func IsNotEmpty[T any](s []T) bool {
+	return !IsEmpty(s)
+}
```

Comment:
if we stick with this pattern we'll need to:
- implement more methods for asserting the length of maps
- ensure subsequent changes to the model code use these helper methods in PR review (image an OS contribution making changes to this whose unaware of these helper methods) 


So, I'd prefer we make direct use of the builtin `len()` function, rather than adopt usage of these methods.
Comparing the current usage:
```go
if validate.IsEmpty(r.Values) {
	mErr.Errors = append(mErr.Errors, errors.New("selector values cannot be empty for In or NotIn operators"))
}
```
vs
```go
if len(r.Values) == 0 {
	mErr.Errors = append(mErr.Errors, errors.New("selector values cannot be empty for In or NotIn operators"))
}
```
I find the latter clearer; This just feels overly complicated. But I won't die on this hill, if you feel strongly about it then let's keep it.